### PR TITLE
files loop & embedded filename & validation

### PR DESCRIPTION
**Pull request by mvenghaus**
- added loop for multiple files
- embed orginal filename in tmp name (like livewire)

### CODE REVIEW

File: src/Controller/Post/Upload.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
 use Magento\Framework\Exception\RuntimeException;
 use Magento\Framework\Math\Random;
 use Magento\Framework\Stdlib\DateTime\DateTime;
-use Magento\MediaStorage\Model\File\UploaderFactory as FileUploaderFactory;
+use Magewirephp\Magewire\Model\Upload\File\TemporaryUploaderFactory as TemporaryFileUploaderFactory;
```
Comment (wpoortman) [inline src/Controller/Post/Upload.php:24]: Hi @mvenghaus,

I see what your doing, but can you explain the future benefit of having this new TempFileUploader object?

Comment (mvenghaus) [reply src/Controller/Post/Upload.php:24]: @wpoortman i didn't find a way to get the original file name out of the default uploader

Comment (wpoortman) [reply src/Controller/Post/Upload.php:24]: Check, and why is it required to have the original file name? Just to make it a bit more clean for me.

Comment (mvenghaus) [reply src/Controller/Post/Upload.php:24]: after uploading i only have the hash tmp name .. but i need the original file name to show the user which files has been added
livewire adds the filename with base64 to the hashname .. so i ported the 2 functions from livewire to magewire

Comment (mvenghaus) [reply src/Controller/Post/Upload.php:24]: image

**Review by wpoortman — COMMENTED** [review]
Do you think this is ready to be merged and we just continue from there? Even it's not perfect, we can make it perfect before merging it into main 😉

**Pull request merged**